test/integration/etcd_storage_path_test.go

```diff
@@ -0,0 +1,1249 @@
+	reflect.TypeOf(&apisauthorizationv1beta1.LocalSubjectAccessReview{}): {ephemeral: true},
+	reflect.TypeOf(&apisauthorizationv1beta1.SubjectAccessReview{}):      {ephemeral: true},
+
+	reflect.TypeOf(&userapiv1.Group{}): {
+		stub: &userapiv1.Group{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "group"},
+			Users: userapiv1.OptionalNames{
+				"user1", "user2",
+			},
+		},
+		expectedEtcdPath: "openshift.io/groups/group",
+	},
+	reflect.TypeOf(&userapiv1.User{}): {
+		stub: &userapiv1.User{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "user1"},
+			FullName:   "user1",
+		},
+		expectedEtcdPath: "openshift.io/users/user1",
+	},
+	reflect.TypeOf(&userapiv1.Identity{}): {
+		stub: &userapiv1.Identity{
+			ObjectMeta:       kapiv1.ObjectMeta{Name: "github:user2"},
+			ProviderName:     "github",
+			ProviderUserName: "user2",
+		},
+		expectedEtcdPath: "openshift.io/useridentities/github:user2",
+	},
+	reflect.TypeOf(&userapiv1.UserIdentityMapping{}): {ephemeral: true}, // pointer from user to identity, not stored in etcd
+
+	reflect.TypeOf(&projectapiv1.Project{}):        {ephemeral: true}, // proxy for namespace so cannot test here
+	reflect.TypeOf(&projectapiv1.ProjectRequest{}): {ephemeral: true}, // not stored in etcd
+
+	reflect.TypeOf(&apisimagepolicyv1alpha1.ImageReview{}): {ephemeral: true}, // not stored in etcd
+
+	reflect.TypeOf(&quotaapiv1.ClusterResourceQuota{}): {
+		stub: &quotaapiv1.ClusterResourceQuota{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "quota1"},
+			Spec: quotaapiv1.ClusterResourceQuotaSpec{
+				Selector: quotaapiv1.ClusterResourceQuotaSelector{
+					LabelSelector: &unversioned.LabelSelector{
+						MatchLabels: map[string]string{
+							"a": "b",
+						},
+					},
+				},
+			},
+		},
+		expectedEtcdPath: "openshift.io/clusterresourcequotas/quota1",
+	},
+	reflect.TypeOf(&quotaapiv1.AppliedClusterResourceQuota{}): {ephemeral: true}, // mirror of ClusterResourceQuota that cannot be created
+
+	// not stored in etcd
+	reflect.TypeOf(&securityapiv1.PodSecurityPolicyReview{}):            {ephemeral: true},
+	reflect.TypeOf(&securityapiv1.PodSecurityPolicySelfSubjectReview{}): {ephemeral: true},
+	reflect.TypeOf(&securityapiv1.PodSecurityPolicySubjectReview{}):     {ephemeral: true},
+
+	reflect.TypeOf(&apisstoragev1beta1.StorageClass{}): {
+		stub: &apisstoragev1beta1.StorageClass{
+			ObjectMeta:  kapiv1.ObjectMeta{Name: "sc1"},
+			Provisioner: "aws",
+		},
+		expectedEtcdPath: "kubernetes.io/storageclasses/sc1",
+	},
+
+	// not stored in etcd
+	reflect.TypeOf(&apiscomponentconfigv1alpha1.KubeSchedulerConfiguration{}): {ephemeral: true},
+	reflect.TypeOf(&apiscomponentconfigv1alpha1.KubeletConfiguration{}):       {ephemeral: true},
+	reflect.TypeOf(&apiscomponentconfigv1alpha1.KubeProxyConfiguration{}):     {ephemeral: true},
+
+	reflect.TypeOf(&apisauthenticationv1beta1.TokenReview{}): {ephemeral: true}, // not stored in etcd
+
+	// we cannot create these  // TODO but we should be able to create them in kube
```

Comment:
Our role bindings are not a real object - they are part of the policy object.  This is not true in kube and thus we need to test for them.